src/base/StatBin.cpp

```diff
@@ -48,9 +48,12 @@ StatBin Merge(const StatBin &lhs, const StatBin &rhs) {
     merged.variance_ = 0.;
   }
   // Bootstrap samples
-  if (lhs.sample_means_.size() != rhs.sample_means_.size()) {
+  if (lhs.sum_weight_ == 0.) {
     merged.sample_means_ = rhs.sample_means_;
     merged.sample_weights_ = rhs.sample_weights_;
+  } else if (rhs.sum_weight_ == 0.) {
+    merged.sample_means_ = lhs.sample_means_;
+    merged.sample_weights_ = lhs.sample_weights_;
   } else {
```

Comment:
the only time when the division by sum of weights happens is when one of the bins which are being merged is not empty. so I am not sure why nan is still popping up?

Reply:
> @kreisl, propose: in StatBin::MergeBins() add special case for empty lhs and rhs in the same time, beacause situation improved, but somehow, some of bin errors still are nan.

Can you send me the container before merging? I can have a look and try to debug.

Reply:
Okay, I'm sending you file with containers via email.